### PR TITLE
GEODE-6210: Add 'transaction' example for cpp

### PR DESCRIPTION
**Pull request by moleske**
- demonstrates commit and rollback of transactions
 - handles exceptions

### CODE REVIEW

**Comment by pivotal-jbarrett** [comment]
Something like:
```c++
main() {
  ...
  transactionManager.begin();
  try {
    for (auto& key : batch) {
      auto value = getValueFromExternalSystem(key);
      cache.put(key, value);
    }
    transactionManager.commit();
  } catch ( ... ) {
    transactionManager.rollback();
  }
}

int32_t getValuFromExternalSystem(int32_t key) {
  if (random(10) == 0) {
    throw "failed to get from external system";
  }

  return random(key);
}
```

**Comment by pdxcodemonkey** [comment]
I was getting stuck on the idea that we needed to force an exception external to the example code to have a "real" example.  Charlie has suggested we implement a retry path as well, which also seems like a good suggestion.  Thoughts?

**Comment by pdxcodemonkey** [comment]
Le sigh - is there a modern C++ version that's not way more complex?  STL stuff seems like overkill

File: examples/cpp/transaction/main.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    "Key6",
+    "Key7",
+    "Key8",
+    "Key9",
+    "Key10"
+};
+
+
+int getValueFromExternalSystem() {
+  std::random_device rd{};
```
Comment (pivotal-jbarrett) [inline examples/cpp/transaction/main.cpp]: This compiles? So we have non-static `rd` passed to a static `e`. At best here we allocating more `rd` on each call that we will never use.

Comment (pivotal-jbarrett) [reply examples/cpp/transaction/main.cpp]:
```C++
static thread_local std::default_random_engine generator(std::random_device{}());
auto value = std::uniform_int_distribution<int32_t>{0, 9}(generator);
...
```

**Pull request merged**